micro-rdk/src/common/conn/server.rs

```diff
@@ -305,30 +317,57 @@ where
             app_config,
             app_client: None,
             incoming_connection_manager: IncomingConnectionManager::new(max_concurent_connections),
+            network,
         }
     }
     pub async fn serve(&mut self, robot: Arc<Mutex<LocalRobot>>) {
         let cloned_robot = robot.clone();
         loop {
             let _ = async_io::Timer::after(std::time::Duration::from_millis(300)).await;
-
+            {
+                match TcpStream::connect("app.viam.com:443") {
+                    Ok(_) => {}
+                    Err(err) => {
+                        log::error!("connection to app.viam.com unavailable, waiting before next signaling attempt: {:?}", err);
+                        let _ =
+                            async_io::Timer::after(std::time::Duration::from_millis(1000)).await;
+                        if !self.network.is_connected().unwrap_or(false) {
+                            // the IP may have changed (such as in the case where we are disconnected
+                            // from a Wi-Fi base station), so we want to prompt the creation of a new
+                            // app client
+                            let _ = self.app_client.take();
+                        }
+                        continue;
+                    }
+                };
+            }
```

Comment:
I'll try testing it again, but I found it took much longer for that function to fail than simply pinging app directly (maybe because of the async part?)

Reply:
ok yeah, I've reworked it a little. Let me know what you think.